executor/insert_common.go

```diff
@@ -191,15 +191,13 @@ func (e *InsertValues) insertRows(ctx context.Context, exec func(ctx context.Con
 			return errors.Trace(err)
 		}
 		rows = append(rows, row)
-		if e.rowCount%uint64(batchSize) == 0 {
+		if batchInsert && e.rowCount%uint64(batchSize) == 0 {
```

Comment:
What if `batchInsert ` is `false`?

Reply:
If it is false, all the rows will be handled at 
https://github.com/pingcap/tidb/pull/9406/files#diff-0f31e77a0899889b749ff90c69acaaf6R204